### PR TITLE
Feature/4.2.1/front end filtering

### PR DESCRIPTION
**Pull request by AdelBeit**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

- Refactored tickets table to separate viewTemplates from component
- added filtering functionality for each field

## Related Tickets & Documents

_Link user story from projects.digitalaidseattle.org_

- Related Issue #4.2.1
- Closes #4.2.1.*

## QA Instructions, Screenshots, Recordings

filter every column, try every filter setting

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [X] Semantic HTML implemented?
- [ ] Keyboard operability supported?
- [ ] Color contrast tested? _more info [here](https://webaim.org/articles/contrast/evaluating)_

## Added/updated tests?
_There should be 90% statement coverage and the coverage percent increases with each commit, unless a reviewer approves an exception._

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [X] I need help with writing tests

## Can reviewers merge this PR upon approval?
- [X] yes, reviewers can merge this PR
- [ ] no, the author of this PR needs to merge it

### CODE REVIEW

File: __tests__/src/components/TicketsTable.test.tsx [hunk 1]
```diff
@@ -53,25 +44,9 @@ describe('TicketTable Headers', () => {
...
-      description: 'Grooming service',
-      created_at: '2023-04-15T10:30:00Z',
-      pet: 'Buddy',
-      client: 'Bob Johnson',
-    },
-  ];
-
+  const items = mockServiceRequestSummaries;
+  const row = items[0];
+  const row2 = items[0];
```
Comment (sharrief) [inline __tests__/src/components/TicketsTable.test.tsx:49]: I think you meant row2 = items[1], also you could write as
```typescript
const [row, row2] = mockServiceRequestSummaries;
```

Comment (AdelBeit) [reply __tests__/src/components/TicketsTable.test.tsx:49]: wow I dont even remember the tests i wrote, thanks I will need more help writing tests

**Comment by AdelBeit** [comment]
still working on writing the tests but it pushed me to separate the logic into a hook for testing so here's the updated code, though id push it early for review while i get the tests out

**Comment by AdelBeit** [comment]
it's ready for review

File: __tests__/src/components/TicketsTable.test.tsx [hunk 2]
```diff
@@ -83,18 +78,146 @@ describe('TicketsTable', () => {
...
+      fireEvent.click(filterMenuButton);
+    });
+    // search for other categories 
+    // TODO: the test isn't triggering filtering functionality for some reason, but it works exactly right on UI
+    const expectedSearchResults = screen.getAllByRole('cell', { name: new RegExp(expectedCategory.label, 'i') });
+    // const unexpectedSearchResults = screen.getAllByText(new RegExp(unexpectedCategory.label, 'i'));
+    expect(expectedSearchResults).toHaveLength(1);
+    expect(expectedSearchResults[0]).toBeVisible();
+    // grab all rows and filter invisible ones
+    // bodyRowGroup = screen.getAllByRole('rowgroup')[1];
```
Comment (DillonOLeary) [inline __tests__/src/components/TicketsTable.test.tsx:215]: Did you mean to remove the commented out code?

File: __tests__/src/components/TicketsTable.test.tsx [hunk 3]
```diff
@@ -83,18 +78,146 @@ describe('TicketsTable', () => {
...
+    await userEvent.click(actualOption);
+    console.log(prettyDOM(optionContainer));
+    // check that it's selected
+    expect(optionContainer?.querySelector('svg')).toBeInTheDocument();
+    // close overlay filter menu
+    waitFor(() => {
+      fireEvent.click(filterMenuButton);
+    });
+    // search for other categories 
+    // TODO: the test isn't triggering filtering functionality for some reason, but it works exactly right on UI
```
Comment (DillonOLeary) [inline __tests__/src/components/TicketsTable.test.tsx:209]: Also is this ok? Is the test still measuring the correct function of the code?

File: src/components/TicketsTable/Templates.tsx
Comment (DillonOLeary) [inline src/components/TicketsTable/Templates.tsx:1]: Can you briefly explain what these templates do? I'm not sure I understand

**Review by DillonOLeary — COMMENTED** [review]
Big PR. Thanks for your work on this. I think I have an understanding of what is going on, but if you have time I'd love to set up a call to go over it. I'm not always the best with frontend work

File: src/hooks/useFilters.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  };
+  const filterByName = (request: ServiceRequestSummary) => {
+    const filterValue = externalFilters.owner_and_pet;
+    if (filterValue) {
+      const filteredFieldsContent = [
+        request.client.first_name,
+        request.client.last_name,
+        request.pet.name,
+      ];
+      return filteredFieldsContent.some(
```
Comment (DillonOLeary) [inline src/hooks/useFilters.ts:46]: If the user enters the first and last name into the filter it doesn't seem to work. I think you just need to join the first name and last name search

**Review by DillonOLeary — CHANGES_REQUESTED** [review]
Left a small note. I'm ok approving it with the issue, but I think it might be confusing for the users

**Review by DillonOLeary — APPROVED** [review]
Looks great!!

**Pull request merged**